docs/modules/ROOT/pages/references/architecture/metering-data-flow.adoc

```diff
@@ -0,0 +1,652 @@
+
+Fields that are marked in **bold** are part a single unique constraint group for that table.
+====
+
+At the center of the model is the fact.
+A fact represents a sampled amount of a billed item (such as memory usage for a single namespace).
+The sampling rate for facts is one hour.
+For ingestion into the target, facts belonging to the same billed item are aggregated to a single line item.
+The value of the fact is the observed quantity to be billed.
+
+The query holds the PromQL query string used to generate facts.
+It also holds metadata associated with the fact, for example the unit and the description to be shown for the line item on the invoice.
+The description is a rendered string from a template.
+
+The product holds the amount to be charged per unit.
+Its source identifier is a segmented match as explained in <<System idea>>.
+This is effectively the price table.
+
+The discount holds a percentage to be discounted.
+Its source identifier is a segmented match as explained in <<System idea>>.
+
+The category allows grouping line items together.
+Taking the example of {product}, a category is a namespace on a specific cluster.
+All billed items of that namespace will be grouped together on the resulting invoice.
+
+== Data flow
+
+=== Query phase
+
+In the query phase, the Prometheus API is queried, and the results are written to the facts table.
+If records in the dimension tables are missing, they're created as needed.
+The created dimension records might be incomplete.
+They will get completed in the next phase.
+
+From an orchestration perspective, a master job is created in the desired interval.
+The master job looks at the query table and identifies the queries that apply to the time window at hand.
+For each identified query, a job performing that query is created.
+
+Each query job fetches the data from Prometheus and writes its results into the facts table.
+If the facts table already contains results for the query job's query and timestamp, those records are updated or dropped and replaced with the new results.
+Table locking is required to prevent two query jobs running in parallel from creating the same dimension records.
+
+The execution interval for this phase is in the range of minutes, hours or possibly days.
+
+== Enrichment phase
+
+The query phase may have created dimension records that are incomplete.
+In the enrichment phase, one or several jobs communicate with the target system to create or fetch any missing information in dimension records.
+It's possible that this phase creates new records in the target system.
+For example, in order to get the target id of a category, this category may need to be created in the target system first.
+
+This phase doesn't necessarily need to be executed as frequently the <<_query_phase,query phase>>.
+However, the enrichment must be completed successfully at least once before the <<_ingestion_phase,ingestion phase>> can be run.
```

Comment:
I think you wanted to indicate that enrichment doesn't need to run after every query, so I reworded this paragraph to make that much clearer. I find it hard to think about intervals as "less" or "more" and to an extent "larger" "smaller".